### PR TITLE
Add libxcrypt-compat package for psycopg2 compatibility

### PR DESCRIPTION
**Pull request by ddelabru**
A project using the tox container to run Python unit tests on a
Django app for a GitLab CI pipleine has started to encounter
pipeline failures because the image on quay.io has been updated to
a more recent version of the Fedora base image. The app uses the
psycopg2 Python library to connect to a PostgreSQL database, and
the unit tests fail upon loading psycopg2 because of a shared
library object file (libcrypt.so.1) which is no longer included in
the Fedora base system. In Fedora 30, this file can be provided
through the libxcypt-compat package.

Note that the Django app is using the pscyopg2-binary repository
from PyPI; if we were allowing pip to compile the C components of
pscopg2 entirely from scratch, it would have PostgreSQL-related
build requirements.

### CODE REVIEW

**Comment by ddelabru** [comment]
@cit1zen, we talked about this briefly. Let me know if you have any questions.

**Comment by csomh** [comment]
@ddelabru could you please update the commit message to include a longer explanation why this is needed?

I've tried to search for it in order to understand the usecase, but found only this [bugzilla]. `psycopg2` is not installed in `tox-container`, so why is `libxcrypt-compat` needed?

Please help to understand your usecase better. In general I would be against including anything in `tox-container` which is not strictly related to Python-testing, as the image is already 1G+. I barely can wait to remove Python 2 from it :grimacing:. 

If applications need other things than `tox` for their testing, I would probably go for creating a dedicated image for them based on `tox-container`.

Thanks!

[bugzilla]: https://bugzilla.redhat.com/show_bug.cgi?id=1696459

**Comment by sochotnicky** [comment]
I am going to go out on a limb and say this is because some pip package gets pulled in during testing, but that pip package needs to compile some C parts and those depend on this compat lib. I really do wonder if we can make that image smaller but for now I'll settle for better commit message as Hunor points out

**Comment by ddelabru** [comment]
@sochotnicky is mostly right, though in fact our pipelines are failing when the unit tests _import_ one of the Python dependencies, and not during installation. I had explained all this over on GitLab when I initially opened a merge request on the old repo, but didn't copy it over here.

@csomh I've amended the commit to give it a long description and I've copied the text of that into the initial comment up above.

Basically we've been sucessfully running our unit tests with this container and now we can't because of something that's no longer included in the latest Fedora base images.

**Comment by csomh** [comment]
Thanks for the update @ddelabru!
